subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/WtpFacet.java

```diff
@@ -31,6 +32,14 @@
  * Creates the .settings/org.eclipse.wst.common.project.facet.core.xml file for WTP projects.
  */
 public class WtpFacet extends XmlPersistableConfigurationObject {
+    /**
+     * The id of the Eclipse J2EE Web Library container.
+     *
+     * @since 8.3
+     */
+    @Incubating
+    public static final String WEB_LIBS_CONTAINER = "org.eclipse.jst.j2ee.internal.web.container";
```

Comment:
I would like to avoid adding a public field here if we can avoid it. 

Can we move this to an internal class?

Reply:
I'm a little torn on this.

I understand the desire to avoid adding new public API (even in this limited way of duplicating a constant already available).  However it seems like this is a constant that _should_ remain part of public API accessible from within the `ide` project, not just within the new `ide-plugins` project.  If you were manually editing the Eclipse project files you'd need to understand where and how to use this string, it doesn't only have meaning for the Gradle plugin.

It's hard to tell exactly which `ide` class should contain it if not the plugin class itself, as there aren't many class-level descriptions in this project, but associating it with the web tools facet of a project seems correct to me.  We could move it to an internal class (maybe `WtpComponentFactory`?) but that seems like it would be less coherent and it would make it impossible for any hypothetical consumer of just `ide` to properly reference it.

I'd like to see what the IDE team thinks here before changing it.